common/buildcraft/robotics/ZonePlannerMapDataClient.java

```diff
@@ -0,0 +1,28 @@
+package buildcraft.robotics;
+
+import buildcraft.lib.BCMessageHandler;
+import net.minecraft.world.World;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class ZonePlannerMapDataClient extends ZonePlannerMapData {
+    public static ZonePlannerMapDataClient instance = new ZonePlannerMapDataClient();
+    public List<Pair<Pair<Integer, Integer>, Consumer<ZonePlannerMapChunk>>> pendingRequests = new ArrayList<>();
+
+    @Override
+    public void loadChunk(World world, int chunkX, int chunkZ, Consumer<ZonePlannerMapChunk> callback) {
+        Pair<Integer, Integer> chunkPosPair = Pair.of(chunkX, chunkZ);
+        Pair<Pair<Integer, Integer>, Consumer<ZonePlannerMapChunk>> pendingRequest;
+        Pair[] pendingRequestsArray = new Pair[]{null};
+        pendingRequest = Pair.of(chunkPosPair, zonePlannerMapChunk -> {
+            pendingRequests.remove(pendingRequestsArray[0]);
+            callback.accept(zonePlannerMapChunk);
+        });
+        pendingRequestsArray[0] = pendingRequest;
+        pendingRequests.add(pendingRequest);
+        BCMessageHandler.netWrapper.sendToServer(new MessageZonePlannerMapChunkRequest(chunkX, chunkZ));
+    }
+}
```

Comment:
Instead of ````Pair<Integer, Integer>```` you should use ````ChunkPos```` as that's a better class for it.

You *might* want to consider changing this to
````java
private final Map<ChunkPos, Deque<Consumer<ZonePlannerMapChunk>> requestors = new HashMap<>();
````
Where you add requesters to a ````ConcurrentLinkedDeque<>````, if the chunk doesn't exist. (I say concurrent mostly so that we can look into threading the chunk rendering later, although it could be changed if its just an unnecessary addition).